lib/_emerge/actions.py

```diff
@@ -2839,31 +2848,25 @@ def getportageversion(portdir, _unused, profile, chost, vardb):
     if profilever is None:
         profilever = "unavailable"
 
-    libcver = []
-    libclist = set()
-    for atom in expand_new_virt(vardb, portage.const.LIBC_PACKAGE_ATOM):
-        if not atom.blocker:
-            libclist.update(vardb.match(atom))
-    if libclist:
-        for cpv in sorted(libclist):
-            libc_split = portage.catpkgsplit(cpv)[1:]
-            if libc_split[-1] == "r0":
-                libc_split = libc_split[:-1]
-            libcver.append("-".join(libc_split))
-    else:
-        libcver = ["unavailable"]
+    return profilever
 
+
+def getportageversion(portdir, _unused, profile, chost, vardb):
+    pythonver = (
+        "python"
+        f" {sys.version_info[0]}"
+        f".{sys.version_info[1]}"
+        f".{sys.version_info[2]}"
+        f"-{sys.version_info[3]}"
+        f"-{sys.version_info[4]}"
+    )
+
+    profilever = get_profile_version(portdir, profile, vardb)
+    libcver = get_libc_version(vardb)
     gccver = getgccversion(chost)
     unameout = platform.release() + " " + platform.machine()
 
-    return "Portage {} ({}, {}, {}, {}, {})".format(
-        portage.VERSION,
-        pythonver,
-        profilever,
-        gccver,
-        ",".join(libcver),
-        unameout,
-    )
```

Comment:
Are we gaining anything by refactoring these functions, or just aesthetic?

Reply:
I don't mind this one, it's cleaner the new way, I'd say.

Reply:
Sure, I agree it looks cleaner, I just wasn't sure if I was missing some magic optimization or something

Reply:
The original function was very big. It was hard to read and keep track of. That's why split into three functions in the spirit of "single responsibility". `get_libc_version()` gets libc's version, `get_profile_version()` gets profile's version etc. No functionality or logic is changed.